toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -85,8 +85,25 @@ func CreateInstallRoot(installRoot string, mountPointMap, mountPointToMountArgsM
 	}
 	installMap[rootMountPoint] = mountPointMap[rootMountPoint]
 
-	// Mount rest of the mountpoints
-	for mountPoint, device := range mountPointMap {
+	// Convert the installMap into a slice of mount points so it can be sorted
+	var remainingMounts []string
+	for mountPoint := range mountPointMap {
+		// Skip empty mount points
+		if mountPoint == "" {
+			continue
+		}
+
+		remainingMounts = append(remainingMounts, mountPoint)
+	}
+
+	// We need to make sure we sort the mount points so we don't mount things in the wrong order
+
+	// Mount the rest of the mount points
+	// This way nested mounts will be handled correctly:
+	// e.g.: /dev is mounted before /dev/pts is.
+	sort.Sort(sort.StringSlice(remainingMounts))
+	for _, mountPoint := range remainingMounts {
+		device := mountPointMap[mountPoint]
```

Comment:
Nice improvement, but I'd take it a step further now:
1. Move the map -> sorted slice convertion into a separate function like `orderMounts`.
2. Remove the initial root mount, since now `orderMounts` should make sure that we start with the root anyway.
3. Remove the `if mountPoint != "" && mountPoint != rootMountPoint` check below, since now they shouldn't be necessary.